### PR TITLE
feat(core): add base-entity and user-modifiable-entity mixins

### LINKED ISSUE #814: Mixins to extend UserModifiableEntity and BaseEntity classes
**Is your feature request related to a problem? Please describe.**
As of now, the @sourceloop core package provides a two models - UserModifiableEntity and BaseEntity that can be extended by the user, but the properties in these models are not configurable, exporting them as Mixin would provide an option to make them configurable, as well as Working with other base classes.

**Describe the solution you'd like**
These Models should be available as Mixins.

### PR DESCRIPTION
**Pull request by akshatdubeysf**
GH-814

## Description

Add BaseEntityMixin and UserModifiableEntityMixin to allow configs and decoupled extending of these properties.

Fixes #814 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Checklist:

- [X] Performed a self-review of my own code
- [X] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [X] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/user-onboarding/angular.json [hunk 1]
```diff
@@ -35,21 +35,8 @@
             "tsConfig": "projects/user-onboarding-lib/tsconfig.spec.json",
             "karmaConfig": "projects/user-onboarding-lib/karma.conf.js"
           }
-        },
```
Comment (samarpan-b) [inline packages/user-onboarding/angular.json:38]: why these changes part of this commit ? make it separated

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: packages/user-onboarding/angular.json (hunk 1 above)
Comment (akshatdubeysf) [reply packages/user-onboarding/angular.json]: fixed, needed to rebase from master
